src/main/kotlin/fi/aalto/cs/apluscourses/services/OpenDocumentationActionProvider.kt

```diff
@@ -0,0 +1,67 @@
+package fi.aalto.cs.apluscourses.services
+
+import com.intellij.ide.browsers.actions.OpenInBrowserBaseGroupAction.OpenInBrowserEditorContextBarGroupAction
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.impl.SimpleDataContext
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.markup.InspectionWidgetActionProvider
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.psi.PsiManager
+import fi.aalto.cs.apluscourses.icons.CoursesIcons
+import fi.aalto.cs.apluscourses.services.course.CourseManager
+import fi.aalto.cs.apluscourses.utils.FileUtil
+
+internal class OpenDocumentationActionProvider : InspectionWidgetActionProvider {
+    override fun createAction(editor: Editor): AnAction? {
+        val project = editor.project ?: return null
+        CourseManager.course(project) ?: return null
+        val virtualFile = editor.virtualFile ?: return null
+        if (virtualFile.extension != "scala") return null
+        val fileName = virtualFile.nameWithoutExtension
+        val path = virtualFile.path
+        println(path)
+        println(path.replace("/o1/", "/doc/o1/").substringBeforeLast("/"))
+
+//        val module =
+//            ProjectRootManager.getInstance(project).fileIndex.getModuleForFile(virtualFile) ?: return null
+        val file = FileUtil.findFileInDirectoryStartingWith(
+            path.replace("/o1/", "/doc/o1/").substringBeforeLast("/"), fileName
```

Comment:
It seems this is specific to O1. How it shows for other courses? Add some TODO there in the code, anyway.